src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go

```diff
@@ -169,21 +164,35 @@ type ClusterList struct {
 
 // RedpandaConfig is the definition of the main configuration
 type RedpandaConfig struct {
-	RPCServer     SocketAddress `json:"rpcServer,omitempty"`
-	KafkaAPI      SocketAddress `json:"kafkaApi,omitempty"`
-	AdminAPI      SocketAddress `json:"admin,omitempty"`
-	DeveloperMode bool          `json:"developerMode,omitempty"`
-	TLS           TLSConfig     `json:"tls,omitempty"`
+	RPCServer     SocketAddress      `json:"rpcServer,omitempty"`
+	KafkaAPI      []KafkaAPIListener `json:"kafkaApi,omitempty"`
+	AdminAPI      []AdminAPI         `json:"adminApi,omitempty"`
+	DeveloperMode bool               `json:"developerMode,omitempty"`
 	// Number of partitions in the internal group membership topic
 	GroupTopicPartitions int `json:"groupTopicPartitions,omitempty"`
 }
 
-// TLSConfig configures TLS for Redpanda APIs
-type TLSConfig struct {
-	// Configuration of TLS for Kafka API
-	KafkaAPI KafkaAPITLS `json:"kafkaApi,omitempty"`
+// AdminAPI is configuration of the redpanda Admin API
+type AdminAPI struct {
```

Comment:
How come we have `name` under AdminAPI. I suppose we don't use it anywhere at the moment. Why was it needed for `KafkaAPIListener` again?

Reply:
I've added it at one point. The reasoning was that once we really have multiple - e.g. multiple internal, we should be able to name them. But on a second thought, we could also assign some dummy names internally, maybe the names does not have to be exposed. WDYT?

Reply:
I would vote for generic names and not exposing them in cluster cr.